splinepy/helpme/visualize.py

```diff
@@ -162,264 +240,291 @@ def _vedo_showable(spline):
-                "but and `arrow_data_on` is set.",
-            )
+        raise ValueError(f"No spline_data named ({data_name}) for {spline}.")
 
-        if create_vertices:
-            # prepare corresponding queries
-            if getattr(adapted_adata, "has_locations", False):
-                queries = adapted_adata.locations
-                on = None
-            else:
-                queries = spline.show_options["arrow_data_on"]
-                on = queries
-
-            # bound /  dim check - only for splines
-            if not spline.name.startswith("Multi"):
-                bounds = spline.parametric_bounds
-                if queries.shape[1] != len(bounds[0]):
-                    raise ValueError(
-                        "Dimension mismatch: arrow_data locations-"
-                        f"{queries.shape[1]} / para_dim-{spline.para_dim}."
-                    )
-                # tolerance padding. may still cause issues in splinepy.
-                # in that case, we will have to scale queries.
-                lb_diff = queries.min(axis=0) - bounds[0] + settings.TOLERANCE
-                ub_diff = queries.max(axis=0) - bounds[1] - settings.TOLERANCE
-                if any(lb_diff < 0) or any(ub_diff > 0):
-                    raise ValueError(
-                        f"Given locations for ({adata_name}) are outside the "
-                        f"parametric bounds ({bounds})."
-                    )
-
-            # get arrow
-            adata = spline.spline_data.as_arrow(adata_name, on=on)
-
-            # create vertices that can be shown as arrows
-            loc_vertices = Vertices(spline.evaluate(queries), copy=False)
-            loc_vertices.vertex_data[adata_name] = adata
-
-            # transfer options
-            keys = ("arrow_data_scale", "arrow_data_color", "arrow_data")
-            spline.show_options.copy_valid_options(
-                loc_vertices.show_options, keys
-            )
 
-            # add to primitives
-            gus_primitives["arrow_data"] = loc_vertices
-
-        else:  # sample arrows and append to sampled spline.
-            sampled_spline.vertex_data[
-                adata_name
-            ] = spline.spline_data.as_arrow(adata_name, resolutions=res)
-            # transfer options
-            keys = ("arrow_data_scale", "arrow_data_color", "arrow_data")
-            spline.show_options.copy_valid_options(
-                sampled_spline.show_options, keys
-            )
+def _process_spline_color(spline, sampled_spline, res):
+    """
+    Processes color and scalar field.
+    If both are defined, you will see the color of the scalar field.
+    """
+    # first, get defaults
+    default_lighting = "glossy"
+    if spline.para_dim > 1:
+        default_color = "green"
+        if spline.para_dim == 2 == spline.dim:
+            default_lighting = "off"  # smoothness is obvious for 2P2D
+    else:
+        default_color = "black"
```

Comment:
```suggestion
    default_lighting = "glossy" if spline.dim > 2 else "off"
    default_color = "green" if spline.para_dim > 1 else "black"

```
@jzwar 's silent suggestion